src/librustc/mir/interpret/value.rs

```diff
@@ -210,25 +184,81 @@ pub enum Scalar {
     Ptr(Pointer),
 }
 
-impl<'tcx> Scalar {
-    pub fn undef() -> Self {
-        Scalar::Bits { bits: 0, defined: 0 }
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, RustcEncodable, RustcDecodable, Hash)]
+pub enum ScalarMaybeUndef {
+    Scalar(Scalar),
+    Undef,
+}
+
+impl From<Scalar> for ScalarMaybeUndef {
+    fn from(s: Scalar) -> Self {
+        ScalarMaybeUndef::Scalar(s)
+    }
+}
+
+impl ScalarMaybeUndef {
+    pub fn read(self) -> EvalResult<'static, Scalar> {
```

Comment:
`read` sounds a bit like it might access memory. I don't have a great idea for a better name either, though.

Reply:
it's "accessing" the "memory" inside the `Scalar`. I changed it to `unwrap_or_err`

Reply:
:+1: